lib/cosmos/packets/structure.rb

```diff
@@ -464,21 +464,28 @@ def enable_method_missing
 
     protected
 
+    MUTEX = Mutex.new
+
+    def mutex
+      MUTEX.synchronize do
```

Comment:
Adding:
return @mutex if @mutex
at the beginning would make this efficient.

Reply:
I'm not sure if modifying ivars is thread safe. @headius?